kroxylicious-filters/kroxylicious-sasl-inspection/src/main/java/io/kroxylicious/filters/sasl/inspection/SaslInspectionFilter.java

```diff
@@ -83,19 +83,21 @@ public CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey, RequestHea
             case SASL_AUTHENTICATE -> onSaslAuthenticateRequest(header.requestApiVersion(), header, (SaslAuthenticateRequestData) request, context);
             case SASL_HANDSHAKE -> onSaslHandshakeRequest(header, (SaslHandshakeRequestData) request, context);
             default -> {
-                if (!authenticationRequired || currentState.clientIsAuthenticated()) {
+                if (!currentState.isFinishedSuccessfully()) {
+                    String disposition = currentState.isStarted() ? "completed" :  "attempted";
+                    String outcome = authenticationRequired ? "closing connection with error" : "forwarding request";
+                    LOGGER.atInfo()
```

Comment:
So we now always log _something_ if the client makes a request before SASL has finished, and the message makes it clear what action the filter has taken.